ndcube/ndcube.py

```diff
@@ -366,6 +372,98 @@ def extra_coords(self):
                     "value": self._extra_coords_wcs_axis[key]["value"]}
         return result
 
+    def all_world_coords(self, axes=None):
+        """
+        Returns WCS coordinate values of all pixels for all axes.
+
+        Parameters
+        ----------
+        axes: `int` or `str`, or iterable of `int` or `str`
+            Axis number in numpy ordering or unique substring of world_axis_physical_types
+            of axes for which real world coordinates are desired.
+            Default=None implies all axes will be returned.
+
+        Returns
+        -------
+        axes_coords: `list` of `astropy.units.Quantity`
+            Real world coords for axes in order requested by user.
+
+        Example
+        -------
+        >>> NDCube.all_world_coords(('lat', 'lon')) # doctest: +SKIP
+        >>> NDCube.all_world_coords(2) # doctest: +SKIP
+
+        """
+        # Define the dimensions of the cube and the total number of axes.
+        cube_dimensions = np.array(self.dimensions.value, dtype=int)
+        n_dimensions = len(cube_dimensions)
```

Comment:
probably slightly cleaner to use `cube_dimensions.size` here.

Reply:
Done